File: WitchcraftApp/src/main/java/com/tarbadev/witchcraft/recipes/domain/usecase/LastAddedRecipesUseCase.java
package com.tarbadev.witchcraft.recipes.domain.usecase;

import com.tarbadev.witchcraft.recipes.domain.entity.Recipe;
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class LastAddedRecipesUseCase {
  private final RecipeRepository recipeRepository;

  public LastAddedRecipesUseCase(RecipeRepository recipeRepository) {
    this.recipeRepository = recipeRepository;
  }

  public List<Recipe> execute() {
    return recipeRepository.findLastAddedRecipes();
  }
}

File: WitchcraftApp/src/main/kotlin/com/tarbadev/witchcraft/recipes/domain/usecase/LastAddedRecipesUseCase.kt
package com.tarbadev.witchcraft.recipes.domain.usecase

import com.tarbadev.witchcraft.recipes.domain.entity.Recipe
import com.tarbadev.witchcraft.recipes.domain.repository.RecipeRepository
import org.springframework.stereotype.Component

@Component
class LastAddedRecipesUseCase(private val recipeRepository: RecipeRepository) {

    fun execute(): List<Recipe> {
        return recipeRepository.findLastAddedRecipes()
    }
}
